src/panel-ladder.tsx

```diff
@@ -0,0 +1,182 @@
+	};
+	Help = () => {
+		return <>
+			<p><button name="selectFormat" onClick={this.setShowHelp(false)}><i class="fa fa-chevron-left"></i> Format List</button></p>
+			<h3>How the ladder works</h3>
+			<p>Our ladder displays three ratings: Elo, GXE, and Glicko-1.</p>
+			<p><strong>Elo</strong> is the main ladder rating. It's a pretty normal ladder rating: goes up when you win and down when you lose.</p>
+			<p><strong>GXE</strong> (Glicko X-Act Estimate) is an estimate of your win chance against an average ladder player.</p>
+			<p><strong>Glicko-1</strong> is a different rating system. It has rating and deviation values.</p>
+			<p>Note that win/loss should not be used to estimate skill, since who you play against is much more important than how many times you win or lose. Our other stats like Elo and GXE are much better for estimating skill.</p>
+		</>;
+	};
+	BattleFormatList = () => {
+		if (!BattleFormats) {
+			return <p>Loading...</p>;
+		}
+		let currentSection: string = '';
+		let sections: JSX.Element[] = [];
+		let formats: JSX.Element[] = [];
+		for (const [key, format] of Object.entries(BattleFormats)) {
+			if (!format.rated || !format.searchShow) continue;
+			if (format.section !== currentSection) {
+				if (formats.length > 0) {
+					sections.push(<preact.Fragment key={currentSection}><h3>{currentSection}</h3><ul style="list-style:none;margin:0;padding:0">{formats}</ul></preact.Fragment>);
+					formats = [];
+				}
+				currentSection = format.section;
+			}
+			formats.push(<li key={key} style="margin:5px"><button name="selectFormat" value={key} class="button" style="width:320px;height:30px;text-align:left;font:12pt Verdana" onClick={this.handleSetFormat(key)}>{BattleLog.escapeFormat(format.id)}</button></li>);
+		}
+		return <>
+			{sections}
+		</>;
+	};
+	ShowFormatList = () => {
+		return <>
+			<p>See a user's ranking with <a class="button" href={`/${Config.routes.users}/`} target="_blank">User lookup</a></p>
+			<this.Notice/>
+			<p>(btw if you couldn't tell the ladder screens aren't done yet; they'll look nicer than this once I'm done.)</p>
+			<p><button name="selectFormat" value="help" class="button" onClick={this.setShowHelp(true)}><i class="fa fa-info-circle"></i> How the ladder works</button></p>
+			<this.BattleFormatList/>
+		</>;
+	};
+	FormatListButton = () => {
+		return <button name="selectFormat" onClick={this.handleSetFormat(undefined)}><i class="fa fa-chevron-left"></i> Format List</button>;
+	};
+	ShowFormat = () => {
+		const { teams } = PS;
+		const { room } = this.props;
+		const { searchValue, lastSearch } = this.state;
+		const selectedFormat = room.selectedFormat as string;
+		if (room.ladderData === undefined) {
+			return <div class="ladder pad"><p><this.FormatListButton/></p><p><em>Loading...</em></p></div>;
+		} else if (teams.usesLocalLadder) {
+			return <div class="ladder pad">
+				<p><this.FormatListButton/></p>
+				<div dangerouslySetInnerHTML={{__html: room.ladderData}}></div>
```

Comment:
Isn't this normally sanitized with `sanitizeHTML`?

Reply:
We might even want to make a component sanitizing HTML, if it doesn't already exist.